### PR TITLE
Update ICONs as per Cass's work

### PR DESCRIPTION
**Pull request by jdunkerley**
### Pull Request Description

Apply the Date and Time based ICON updates.

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [ ] The documentation has been updated, if necessary.
- [ ] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [ ] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [ ] Unit tests have been written where possible.
  - [ ] If GUI codebase was changed, the GUI was tested when built using `./run ide build`.

### CODE REVIEW

**Review by AdRiley — APPROVED** [review]

File: distribution/lib/Standard/Base/0.0.0-dev/src/Data/Time/Duration.enso [hunk 1]
```diff
@@ -65,7 +65,8 @@ ensure_duration object ~action =
 ## Represents the amount of time between two points in time.
 @Builtin_Type
 type Duration
-    ## Create an interval representing the duration between two points in time.
+    ## ICON operators
+       Create an interval representing the duration between two points in time.
```
Comment (radeusgd) [inline distribution/lib/Standard/Base/0.0.0-dev/src/Data/Time/Duration.enso]: Given that `up_to` and `down_to` on `Date_Range` got the `date_and_time` icon, why shouldn't `Duration.between` be the same?

File: distribution/lib/Standard/Base/0.0.0-dev/src/Data/Time/Time_Of_Day.enso [hunk 2]
```diff
@@ -421,6 +432,7 @@ type Time_Of_Day
             self.format "HH:mm:ss.ffffff"
 
     ## GROUP Conversions
+       ICON text
```
Comment (radeusgd) [inline distribution/lib/Standard/Base/0.0.0-dev/src/Data/Time/Time_Of_Day.enso]: I'm wondering if `convert` wouldn't fit here better?

File: distribution/lib/Standard/Base/0.0.0-dev/src/Data/Time/Time_Zone.enso [hunk 3]
```diff
@@ -50,7 +50,8 @@ parse_builtin text = @Builtin_Method "Time_Zone.parse_builtin"
    "Europe/Paris".
 @Builtin_Type
 type Time_Zone
-    ## The system default timezone.
+    ## ICON date_and_time
+       The system default timezone.
 
        > Example
          Get the system default timezone.
```
Comment (radeusgd) [inline distribution/lib/Standard/Base/0.0.0-dev/src/Data/Time/Time_Zone.enso:57]: Completely unrelated to this PR, but looking at the examples, I'm wondering if such trivial ones should be kept - given the component browser allows to easily create a node like the one shown here and the example description is just repeating the method description - it seems rather redundant. I guess such simple methods maybe shouldn't have examples at all? It feels like they only make sense if we can show some interesting usage.

**Review by radeusgd — APPROVED** [review]
Looks good, just a few minor comments.

File: distribution/lib/Standard/Base/0.0.0-dev/src/Data/Time/Duration.enso (hunk 1 above)
Comment (jdunkerley) [reply distribution/lib/Standard/Base/0.0.0-dev/src/Data/Time/Duration.enso]: Moved to `date_time`.

File: distribution/lib/Standard/Base/0.0.0-dev/src/Data/Time/Time_Of_Day.enso (hunk 2 above)
Comment (jdunkerley) [reply distribution/lib/Standard/Base/0.0.0-dev/src/Data/Time/Time_Of_Day.enso]: done.

**Review by Cassandra-Clark — APPROVED** [review]

**Review by GregoryTravis — APPROVED** [review]

**Pull request merged**